seed/utils/filter_state.py

```diff
@@ -0,0 +1,175 @@
+"""
+:copyright (c) 2014 - 2022, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
+:author
+"""
+
+from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
+from django.db.utils import DataError
+from django.http import JsonResponse
+from rest_framework import status
+from rest_framework.request import Request
+from seed.lib.superperms.orgs.models import Organization
+from seed.models import (VIEW_LIST, VIEW_LIST_PROPERTY, VIEW_LIST_TAXLOT,
+                         Column, ColumnListProfile, ColumnListProfileColumn,
+                         Cycle, PropertyView)
+from seed.models import TaxLotProperty, TaxLotView
+from seed.search import build_view_filters_and_sorts, FilterException
+from seed.serializers.pint import (apply_display_unit_preferences)
+from typing import Union, Optional, Literal
+
+
+def _get_filtered_results(request: Request, state_type: Union[Literal['property'], Literal['taxlot'], None], profile_id: int):
```

Comment:
The typing should be `state_type: str` however mypy threw errors. I'll look into this as it's not clear at the moment.